Sources/ArcGISToolkit/FloorFilter/SiteFacilityPromptViewController.swift

```diff
@@ -0,0 +1,272 @@
+            backBtn?.isHidden = true
+        }
+    }
+}
+
+/// Extension for Search Bar functions
+extension SiteFacilityPromptViewController: UISearchBarDelegate {
+    private func initializeSiteFacilitySearchBar() {
+        siteFacilitySearchBar.delegate = self
+        resetSearchFilteredResults()
+        UIBarButtonItem.appearance(whenContainedInInstancesOf: [UISearchBar.self]).setTitleTextAttributes([NSAttributedString.Key(rawValue: NSAttributedString.Key.foregroundColor.rawValue): UIColor.customBlue], for: .normal)
+    }
+    
+    public func searchBarTextDidBeginEditing(_ searchBar: UISearchBar) {
+        isSearchActive = true
+        self.siteFacilitySearchBar.showsCancelButton = true
+    }
+    
+    public func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        isSearchActive = false
+        self.siteFacilitySearchBar.resignFirstResponder()
+        self.siteFacilitySearchBar.endEditing(true)
+    }
+    
+    public func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
+        isSearchActive = false
+        self.siteFacilitySearchBar.text = ""
+        self.siteFacilitySearchBar.resignFirstResponder()
+        self.siteFacilitySearchBar.endEditing(true)
+    }
+    
+    public func searchBarTextDidEndEditing(_ searchBar: UISearchBar) {
+        isSearchActive = false
+        self.siteFacilitySearchBar.resignFirstResponder()
+        self.siteFacilitySearchBar.endEditing(true)
+    }
+    
+    private func resetSearchFilteredResults() {
+        filteredSearchSites = viewModel.sites
+        filteredSearchFacilities = viewModel.facilities
+    }
+    
+    private func dismissSearchBar() {
+        self.isSearchActive = false
+        self.siteFacilitySearchBar.text = ""
+        self.siteFacilitySearchBar.resignFirstResponder()
+        self.siteFacilitySearchBar.endEditing(true)
+        self.siteFacilitySearchBar.showsCancelButton = false
+    }
+}
+
+/// Extension for the Sites and Facilities Table View
+extension SiteFacilityPromptViewController: UITableViewDataSource, UITableViewDelegate {
+    private func initializeSiteFacilityTableView() {
+        siteFacilityTableView.delegate = self
+        siteFacilityTableView.dataSource = self
+        siteFacilityTableView.separatorStyle = .none
+        siteFacilityTableView.register(UINib(nibName: "FloorFilterSiteFacilityCell", bundle: .module), forCellReuseIdentifier: "FloorFilterSiteFacilityCell")
+    }
+    
+    public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        updatePromptTitle()
+        return isShowingFacilities ? getFilteredFacilities().count : getFilteredSites().count
+    }
+    
+    public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        if let cell = siteFacilityTableView.dequeueReusableCell(withIdentifier: "FloorFilterSiteFacilityCell", for: indexPath) as? SiteFacilityTableViewCell {
```

Comment:
This should be force unwrapped because if it is `nil`, that is a fatal design flaw that needs to be fixed right away. `tableView(_:cellForRowAt:)` should never attempt to fail gracefully by returning a cell that wasn't dequeued.